src/storage/FirebaseStorage.js

```diff
@@ -0,0 +1,92 @@
+(function() {
+  "use strict";
+
+  function unwrapStorageSnapshot(storageSnapshot) {
```

Comment:
Style nit: You have `_assertStorageRef()` below, but don't use a leading underscore in front of `unwrapStorageSnapshot()` and `isStorageRef()`. Let's put a leading underscore in front of each of them for consistency.